src/Random/InverseTransformSampling.h

```diff
@@ -46,36 +46,161 @@ namespace ippl {
         template <class RegionLayout>
         InverseTransformSampling(Distribution &dist_r, Vector<T, Dim> &rmax_r, Vector<T, Dim> &rmin_r, RegionLayout& rlayout_r, size_type &ntotal_r)
         : dist_m(dist_r),
-        rmax_m(rmax_r),
-        rmin_m(rmin_r),
         ntotal_m(ntotal_r){
-            const typename RegionLayout::host_mirror_type regions = rlayout_r.gethLocalRegions();
-            
+
+            updateBounds(rmax_r, rmin_r, rlayout_r);
+
+        }
+
+        /*!
+         * @brief Constructor for InverseTransformSampling class without applying domain decomposition..
+         *
+         * @param dist_ The distribution to sample from.
+         * @param rmax_ Maximum range for sampling.
+         * @param rmin_ Minimum range for sampling.
+         * @param rlayout The region layout.
```

Comment:
I think these comments are wrong (seems copy pasted from the other one without change). Also why it is mentioned as rmin_ and not as rmin_r in the comments?

Reply:
yes. they were copied pasted. I'll update them.